### PR TITLE
Ubuntu Jammy support.

### LINKED ISSUE #94: No jammy (22.04 LTS) support
The charm doesn't support jammy (22.04 LTS) yet. This is not a request of upgrading the server side of the LMA stack, but to enable log aggregation of jammy workload like OpenStack Yoga on 22.04 LTS.

```
$ juju deploy --series jammy filebeat
ERROR series "jammy" not supported by charm, supported series are: focal, bionic, xenial, trusty. Use --force to deploy the charm anyway.
```

When we used `--force` for testing, the install hook fails with:
```
unit-filebeat-0: 07:58:17 DEBUG unit.filebeat/0.install Processing ./wheelhouse/Tempita-0.5.2.tar.gz
unit-filebeat-0: 07:58:17 DEBUG unit.filebeat/0.install   Preparing metadata (setup.py): started
unit-filebeat-0: 07:58:17 DEBUG unit.filebeat/0.install   Preparing metadata (setup.py): finished with status 'error'
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install   error: subprocess-exited-with-error
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install   
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install   × python setup.py egg_info did not run successfully.
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install   │ exit code: 1
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install   ╰─> [1 lines of output]
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install       error in Tempita setup command: use_2to3 is invalid.
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install       [end of output]
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install   
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install   note: This error originates from a subprocess, and is likely not a problem with pip.
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install error: metadata-generation-failed
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install 
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install × Encountered error while generating package metadata.
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install ╰─> See above for output.
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install 
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install note: This is an issue with the package mentioned above, not pip.
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install hint: See above for details.
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install Traceback (most recent call last):
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install   File "/var/lib/juju/agents/unit-filebeat-0/charm/hooks/install", line 8, in <module>
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install     basic.bootstrap_charm_deps()
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install   File "/var/lib/juju/agents/unit-filebeat-0/charm/lib/charms/layer/basic.py", line 206, in bootstrap_charm_deps
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install     check_call([pip, 'install', '-U', reinstall_flag, '--no-index',
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install   File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install     raise CalledProcessError(retcode, cmd)
unit-filebeat-0: 07:58:17 WARNING unit.filebeat/0.install subprocess.CalledProcessError: Command '['/var/lib/juju/agents/unit-filebeat-0/.venv/bin/pip', 'install', '-U', '--force-reinstall', '--no-index', '--no-cache-dir', '-f', 'wheelhouse', 'MarkupSafe', 'charms.templating.jinja2', 'six', 'PyYAML', 'Jinja2', 'pbr', 'pyaml', 'charms.reactive', 'netaddr', 'Tempita', 'wheel', 'charmhelpers']' returned non-zero exit status 1.
```

[filebeat-0.log](https://github.com/juju-solutions/layer-filebeat/files/9024862/filebeat-0.log)

### PR DESCRIPTION
**Pull request by mkalcok**
**Dependency:** [This change to `layer:beats-base`](https://github.com/juju-solutions/layer-beats-base/pull/41) has to be merge as well for proper `Ubuntu 22.04` support (more info in the PR).

This change adds support for deployment of `filebeats` charm on `Ubuntu 22.04` and adds `charmcraft.yaml` file for build compatibility with `charmcraft`.

Two main changes were needed:
* Drop dependency on [charms.templating.jinja2](https://github.com/juju-solutions/charms.templating.jinja2) which depends on [Tempita](https://pypi.org/project/Tempita/) package that does not work on `python3.10` (More info in PR linked above)
* Add `clone3` syscall to list of allowed syscalls. Absence of this syscall caused an issues that are described in [this Github Issue](https://github.com/elastic/apm-server/issues/6238) (It's from `apm` project, not `filebeats` but the problem and the solution are the same)

Resolves #94

### CODE REVIEW

**Review by Pjack — APPROVED** [review]
LGTM

**Comment by Pjack** [comment]
Please also check if there is any useful code in this [PR](https://github.com/juju-solutions/layer-filebeat/pull/91/files) provided by Alvaro before. I will drop his PR after merge this one.

**Comment by mkalcok** [comment]
I linked #94 issue and added Alvaro's changes to the `README.md`

File: templates/filebeat-6.yml
Comment (agileshaw) [inline templates/filebeat-6.yml:141]: As we already supports filebeat 7, could you please check if this syscall is also needed for `filebeat-7.yaml`? You can test it by replacing the default value of [`install_sources`](https://charmhub.io/filebeat/configure#install_sources) config option with `"deb https://artifacts.elastic.co/packages/7.x/apt stable main"`

**Review by agileshaw — CHANGES_REQUESTED** [review]
Thanks for the PR. I've made a suggestion in the inline comment.

File: templates/filebeat-6.yml
Comment (mkalcok) [reply templates/filebeat-6.yml:141]: Yes, I wanted to mention this as well. Filebeat7 has this problem fixed in [https://www.elastic.co/guide/en/beats/libbeat/master/release-notes-7.15.2.html](7.15.2)

**Review by kwmonroe — APPROVED** [review]

**Comment by kwmonroe** [comment]
This LGTM, but i'm curious if you need `jammy` listed in the metadata.yaml:

https://github.com/juju-solutions/layer-filebeat/blob/master/metadata.yaml

I presume that charmcraft and the pack/push procedure means you don't, but wanted to double check just in case it was an oversight.

**Comment by agileshaw** [comment]
@kwmonroe Thanks for the review. The series in `metadata.yaml` is deprecated if using `charmcraft`. The best practice would be to replace the line with `series: []`. But since we are planning to refactoring the charm in near future, I think we can leave it as is right now and save the change in a new PR.

**Pull request merged**
